packages/rest-api-client/src/KintoneRestAPIClient.ts

```diff
@@ -142,3 +144,14 @@ const validateGuestSpaceId = (guestSpaceId: Options["guestSpaceId"]) => {
     throw new Error(`invalid guestSpaceId: got [${guestSpaceId}]`);
   }
 };
+
+const validateSocketTimeout = (socketTimeout: Options["socketTimeout"]) => {
+  if (socketTimeout === undefined) {
+    return;
+  }
+
+  const number = parseFloat(socketTimeout.toString());
+  if (isNaN(number) || socketTimeout < 0) {
+    throw new Error(`Invalid socketTimeout. Must be a positive number.`);
+  }
+};
```

Comment:
Can I ask you what you validate with this block?
I thought we only needed to check `socketTimeout < 0` to check if positive number.

Is it for non-TypeScript users who can specify non-number values?
If so, I think it's great! But there are some points that can be improved.
- MUST: Please use check `number < 0` rather than the original `socketTimeout` (because it can be non-number)
- IMO: add `string` to the type of input to make it clear what we want to guard with this validation (or use `unknown` to make it more strict)

```suggestion
const validateSocketTimeout = (socketTimeout: Options["socketTimeout"] | string) => {
  if (socketTimeout === undefined) {
    return;
  }

  const number = parseFloat(socketTimeout.toString());
  if (isNaN(number) || number < 0) {
    throw new Error(`Invalid socketTimeout. Must be a positive number.`);
  }
};
```

Reply:
@mshrtsr 

> Is it for non-TypeScript users who can specify non-number values?

Yes, correct. We should validate the number format too.
When I check with non-Typescript example and specify `socketTimeout` as a string, the process will hang and there is no notice.

For the negative number, there also has an error message about this. But, It's not clear, so It's hard for the user to know what going on, so I added the validation for the positive number as well.

```
node:internal/timers:396
    throw new ERR_OUT_OF_RANGE(name, 'a non-negative finite number', msecs);
    ^

RangeError [ERR_OUT_OF_RANGE]: The value of "msecs" is out of range. It must be a non-negative finite number. Received -100
```

> Please use check number < 0 rather than the original socketTimeout

Oh, this is my mistake. Thank you.
It should be `number < 0`, not `socketTimeout`

> add string to the type of input to make it clear what we want to guard with this validation

Great. I will update it.